### PR TITLE
feat: add read API

### PR DESCRIPTION
**Pull request by andrewrynhard**
This adds an API for reading arbitrary files.

### CODE REVIEW

File: api/machine/machine.proto [hunk 1]
```diff
@@ -262,3 +263,7 @@ message LogsRequest {
   // driver might be default "containerd" or "cri"
   common.ContainerDriver driver = 3;
 }
+
+message ReadRequest {
+  string path = 1;
```
Comment (Ulexus) [inline api/machine/machine.proto]: Not for the first-pass feature, but properties of `Offset` and `Limit` would be useful from an API perspective.

File: internal/app/machined/internal/api/reg/reg.go [hunk 2]
```diff
@@ -507,3 +508,32 @@ func getContainerInspector(ctx context.Context, namespace string, driver common.
...
+			Response: []*common.DataResponse{
+				{
+					Bytes: &common.Data{Bytes: b},
+				},
+			},
+		}
+
+		return reply, nil
+	default:
+		return nil, fmt.Errorf("path must be a file")
```
Comment (Ulexus) [inline internal/app/machined/internal/api/reg/reg.go]: "regular file", perhaps?  As it stands, things like device nodes, sockets, etc what _are_ files will still get caught by this filter.

All of these are caught by this filter:
```
ModeType = ModeDir | ModeSymlink | ModeNamedPipe | ModeSocket | ModeDevice | ModeCharDevice | ModeIrregular
```

File: api/machine/machine.proto (hunk 1 above)
Comment (andrewrynhard) [reply api/machine/machine.proto]: Agree. Was debating if this should actually be a stream but yeah we can take a closer look in the future.

Comment (Ulexus) [reply api/machine/machine.proto]: +1 to streaming _receive_, but this is the request side, anyway.  I suppose there is an argument to offering flow control via a streaming request side interface, but that's even further out, I think.

File: internal/app/machined/internal/api/reg/reg.go (hunk 2 above)
Comment (andrewrynhard) [reply internal/app/machined/internal/api/reg/reg.go]: Fixed

**Comment by andrewrynhard** [comment]
> So I guess the question, then, is whether you want to make the receive be a streamed interface now or later, with consideration that this is in the public API. Without streaming, you run a bit of a risk with open-endedly-large files.

Proper thing probably is to stream. I'll do that.

**Comment by smira** [comment]
:+1: With streaming API, one can probably read also some non-regular files as well.

File: internal/app/machined/internal/api/reg/reg.go [hunk 3]
```diff
@@ -507,3 +507,38 @@ func getContainerInspector(ctx context.Context, namespace string, driver common.
...
+		f, err := os.Open(in.Path)
+		if err != nil {
+			return err
+		}
+
+		ctx, cancel := context.WithCancel(context.Background())
+
+		defer cancel()
+
+		chunker := stream.NewChunker(f)
```
Comment (smira) [inline internal/app/machined/internal/api/reg/reg.go]: that is not obvious, but `f` is going to be closed by `chunker`

File: internal/app/machined/internal/api/reg/reg.go [hunk 4]
```diff
@@ -507,3 +507,38 @@ func getContainerInspector(ctx context.Context, namespace string, driver common.
...
+
+		chunkCh := chunker.Read(ctx)
```
Comment (smira) [inline internal/app/machined/internal/api/reg/reg.go]: that is more of a question for `stream.Chunker`, but there's no way to get read errors out of it, probably worth an issue to fix later, like providing extra `errCh` back

Comment (andrewrynhard) [reply internal/app/machined/internal/api/reg/reg.go]: Yeah we can and should improve this later. We're doing a lot more streaming now.

File: internal/app/machined/internal/api/reg/reg.go (hunk 3 above)
Comment (andrewrynhard) [reply internal/app/machined/internal/api/reg/reg.go]: Hmm yeah, I'm not seeing _where_ though.

Comment (smira) [reply internal/app/machined/internal/api/reg/reg.go]: should be https://github.com/talos-systems/talos/blob/master/pkg/chunker/stream/stream.go#L65

Comment (andrewrynhard) [reply internal/app/machined/internal/api/reg/reg.go]: Ahh I'm using the file chunker. That one doesn't seem to close the file.

Comment (andrewrynhard) [reply internal/app/machined/internal/api/reg/reg.go]: Should I use the file one? Been a while since I looked at these.

**Pull request merged**